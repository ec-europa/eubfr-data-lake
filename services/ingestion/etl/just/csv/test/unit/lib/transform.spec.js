/**
 * @jest-environment node
 */

import mapper from '../../../src/lib/transform';
import testRecord from '../../stubs/record.json';

describe(`DG JUST CSV transformer`, () => {
  let result = {};

  beforeAll(() => {
    result = mapper(testRecord);
  });

  test(`Produces correct JSON output structure`, () => {
    expect(result).toMatchSnapshot();
  });
});
