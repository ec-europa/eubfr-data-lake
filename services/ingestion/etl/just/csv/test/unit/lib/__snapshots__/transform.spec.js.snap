// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DG JUST CSV transformer Produces correct JSON output structure 1`] = `
Object {
  "action": "",
  "budget": Object {
    "eu_contrib": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "funding_area": Array [],
    "mmf_heading": "",
    "other_contrib": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "private_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "public_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "total_cost": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
  },
  "call_year": "2013",
  "description": "The GEAR against IPV Approach (Gender Equality Awareness Raising against Intimate Partner Violence) was a coordinated action, that took place in 5 countries, namely: Croatia, Cyprus, Greece, Romania and Spain, to prevent primary and secondary Intimate Partner Violence (IPV) in adolescentsâ€™ relationships through child-centred interventions in the school or other setting that were guided by specially designed educational material. The main aim was to promote the development of healthy and equal relationships and the development of zero tolerance towards violence by raising awareness on the characteristics of healthy and unhealthy relationships, on the influence that gender stereotypical attitudes and socially imposed gender roles have on relationships and on how power inequality between the sexes is related to psychological, physical and/or sexual abuse against women/girls.",
  "ec_priorities": Array [],
  "media": Array [],
  "programme_name": "",
  "project_id": "JUST/2013/DAP/AG/5408",
  "project_locations": Array [
    Object {
      "address": "",
      "centroid": null,
      "country_code": "",
      "location": null,
      "nuts": Array [],
      "postal_code": "",
      "region": "",
      "town": "",
    },
  ],
  "project_website": "http://www.gear-ipv.eu",
  "related_links": Array [
    Object {
      "label": "",
      "url": "",
    },
  ],
  "reporting_organisation": "JUST",
  "results": Object {
    "available": "",
    "result": "",
  },
  "status": "",
  "sub_programme_name": "",
  "success_story": "",
  "themes": Array [],
  "third_parties": Array [],
  "timeframe": Object {
    "from": null,
    "from_precision": "year",
    "to": null,
    "to_precision": "year",
  },
  "title": "Gender Equality Awareness Raising against Intimate Partner Violence - II (GEAR against IPV - II)",
  "type": Array [],
}
`;
